node/impl/full/f3.go

```diff
@@ -0,0 +1,57 @@
+package full
+
+import (
+	"context"
+	"errors"
+
+	"go.uber.org/fx"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-f3/certs"
+
+	"github.com/filecoin-project/lotus/chain/lf3"
+)
+
+type F3API struct {
+	fx.In
+
+	F3 *lf3.F3 `optional:"true"`
+}
+
+var ErrF3Disabled = errors.New("f3 is disabled")
+
+func (f3api *F3API) F3Participate(ctx context.Context, miner address.Address) (<-chan string, error) {
+
+	if f3api.F3 == nil {
```

Comment:
I'm having a hard time figuring out where this is initialised. When will this be non nil ?

Reply:
Ah it's optional and will not be set if we don't have an instance of `lf3.F3` available which will happen if `build.F3Enabled` is false.

Reply:
Is that correct ?

Reply:
yes